packages/@aws-cdk/aws-events-targets/lib/util.ts

```diff
@@ -45,3 +46,39 @@ export function addLambdaPermission(rule: events.IRule, handler: lambda.IFunctio
     });
   }
 }
+
+
+/**
+ * Allow a rule to send events with failed invocation to an Amazon SQS queue.
+ */
+export function addToDeadLetterQueueResourcePolicy(rule: events.IRule, queue: sqs.IQueue) {
+  const ruleParsedStack = Stack.of(rule);
```

Comment:
Instead of the `Stack.region`, you should be using `rule.env.region` (and same for account).

This will properly account for the account and region of imported resources.